modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java

```diff
@@ -242,8 +243,8 @@ private enum CommentResolution {
   private IndexService indexService;
   private IngestService ingestService;
   private SecurityService securityService;
-  private CommonEventCatalogUIAdapter eventCatalogUIAdapter;
   private final List<EventCatalogUIAdapter> catalogUIAdapters = new ArrayList<>();
+  private final HashMap<String, List<EventCatalogUIAdapter>> orgCatalogUIAdaptersMap = new HashMap<>();
```

Comment:
```suggestion
  private final List<EventCatalogUIAdapter> catalogUIAdapters = new CopyOnWriteArrayList<>();
  private final HashMap<String, List<EventCatalogUIAdapter>> orgCatalogUIAdaptersMap = new ConcurrentHashMap<>();
```
Many implementations are using a thread-safe list in combination with multiple, dynamic references https://enroute.osgi.org/FAQ/300-declarative-services.html. I have once read an article about why you should that but I cannot find it anymore.